Sources/App/Commands/Analyzer.swift

```diff
@@ -83,26 +83,33 @@ func analyze(application: Application, packages: [Package]) -> EventLoopFuture<V
         }
     }
     
-    let packageUpdates = refreshCheckouts(application: application, packages: packages)
+    let packages = refreshCheckouts(application: application, packages: packages)
         .flatMap { updateRepositories(application: application, packages: $0) }
     
-    let versionUpdates = packageUpdates.flatMap { packages in
-        application.db.transaction { tx -> EventLoopFuture<[Result<Package, Error>]> in
-            let versions = reconcileVersions(client: application.client,
-                                             logger: application.logger,
-                                             threadPool: application.threadPool,
-                                             transaction: tx,
-                                             packages: packages)
-            return versions
-                .map { getManifests(logger: application.logger, versions: $0) }
-                .flatMap { updateVersionsAndProducts(on: tx, packages: $0) }
-                .flatMap { updateLatestVersions(on: tx, packages: $0) }
+    let packageResults = packages.flatMap { packages in
+        application.db.transaction { tx in
+            diffVersions(client: application.client,
+                         logger: application.logger,
+                         threadPool: application.threadPool,
+                         transaction: tx,
+                         packages: packages)
+                .flatMap { applyVersionDelta(on: tx, packageDeltas: $0) }
+                .map { getManifests(logger: application.logger, packageAndVersions: $0) }
+                .flatMap { updateVersions(on: tx, packageResults: $0) }
+                .flatMap { createProducts(on: tx, packageResults: $0) }
+                .flatMap { updateLatestVersions(on: tx, packageResults: $0) }
+                .flatMap { onNewVersions(client: application.client,
+                                         logger: application.logger,
+                                         transaction: tx,
+                                         packageResults: $0)}
```

Comment:
The main point of this exercise (apart from some tidying up): This refactor allows us to place `onNewVersions` at the end of the pipeline, with access to updated properties like `latest`. This will allow us to suppress versions that aren't the latest.